### PR TITLE
Endpoints Pays done

### PR DESCRIPTION
**Pull request by MistraForest**
All Tests were working well, please ckeckt it

### CODE REVIEW

File: backend/src/main/java/com/ambacam/service/PaysService.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+		if (!found.getNom().equals(update.getNom())) {
+			checkConsistency(update);
+		}
+
+		update.setId(id);
+		return paysRepository.save(update);
+	}
+
+	/**
+	 * Delete an pays
```
Comment (Chombon) [inline backend/src/main/java/com/ambacam/service/PaysService.java]: Typo: a pays not an pays

File: backend/src/main/java/com/ambacam/service/PaysService.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+	 * Update an pays
+	 * 
+	 * @param id The id of the pays to update
+	 * 
+	 * @param update The new pays modifications
+	 * 
+	 * @return The pays updated
+	 * 
+	 * @throws ResourceNotFoundException if the pays is not found
+	 * @throws ResourceBadRequestException if an pays with the name already exist
```
Comment (Chombon) [inline backend/src/main/java/com/ambacam/service/PaysService.java]: Typo: a pays not an pays

File: backend/src/main/java/com/ambacam/service/PaysService.java [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+		checkConsistency(pays);
+
+		pays.setId(null);
+
+		return paysRepository.save(pays);
+	}
+
+	/**
+	 * Get an pays
```
Comment (Chombon) [inline backend/src/main/java/com/ambacam/service/PaysService.java]: Typo: a pays not an pays

File: backend/src/main/java/com/ambacam/rest/pays/PaysResource.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+	public Pays get(@PathVariable("paysId") Long paysId) {
+		log.info("Get a pays [id={}]", paysId);
+
+		return paysService.get(paysId);
+
+	}
+
+	@RequestMapping(method = RequestMethod.DELETE)
+	public void delete(@PathVariable("paysId") Long paysId) {
+		log.info("Delete an pays [id={}]", paysId);
```
Comment (Chombon) [inline backend/src/main/java/com/ambacam/rest/pays/PaysResource.java]: Typo: a pays not an pays

File: backend/src/main/java/com/ambacam/repository/PaysRepository.java [hunk 5]
```diff
@@ -0,0 +1,13 @@
+package com.ambacam.repository;
+
+import com.ambacam.model.Pays;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+import javax.inject.Named;
+
+@Named
```
Comment (Chombon) [inline backend/src/main/java/com/ambacam/repository/PaysRepository.java]: Please remove @named

File: backend/src/main/java/com/ambacam/model/Pays.java [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+	private static final long serialVersionUID = -8019398584857375323L;
+
+	@Id
+	@GeneratedValue(strategy = GenerationType.AUTO)
+	private Long id;
+
+	@NotEmpty(message = "The name must not be empty")
+	@NotNull(message = "The name must not be null")
+	@Column(columnDefinition = "VARCHAR", nullable = false)
```
Comment (Chombon) [inline backend/src/main/java/com/ambacam/model/Pays.java]: Please change from VARCHAR to varchar

File: backend/src/test/java/com/ambacam/rest/pays/PayssResourceIT.java [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+				.then()
+				.log().body()
+				.statusCode(200)
+				.extract().body().path("id");
+
+		// check that the pays has been saved
+		Pays actual = repository.findOne(Integer.toUnsignedLong(id));
+		assertThat(actual, is(notNullValue()));
+		assertThat(actual.getNom(), is(equalTo(create.getNom())));
+	}
```
Comment (Chombon) [inline backend/src/test/java/com/ambacam/rest/pays/PayssResourceIT.java]: Please check all properties

File: backend/src/test/java/com/ambacam/rest/pays/PayssResourceIT.java [hunk 8]
```diff
@@ -0,0 +1,189 @@
...
+		given().get(ApiConstants.PAYS_COLLECTION)
+				.then()
+				.log().body().statusCode(200)
+				.body("size()", is(equalTo(2)))
+				.body(
+						"id"
+						, containsInAnyOrder(
+								pays1.getId().intValue(), pays2.getId().intValue()
+						)
+				);
```
Comment (Chombon) [inline backend/src/test/java/com/ambacam/rest/pays/PayssResourceIT.java]: Please check all properties

File: backend/src/test/java/com/ambacam/rest/pays/PayssResourceIT.java [hunk 9]
```diff
@@ -0,0 +1,189 @@
...
+	}
+
+	@Test
+	public void get() {
+		given().get(ApiConstants.PAYS_ITEM, pays2.getId())
+				.then().log().body().statusCode(200)
+				.body("id", is(equalTo(pays2.getId().intValue())))
+				.body("nom", is(equalTo(pays2.getNom())));
+	}
```
Comment (Chombon) [inline backend/src/test/java/com/ambacam/rest/pays/PayssResourceIT.java]: Please check all properties

File: backend/src/test/java/com/ambacam/rest/pays/PayssResourceIT.java [hunk 10]
```diff
@@ -0,0 +1,189 @@
...
+				.body(update).put(ApiConstants.PAYS_ITEM, pays2.getId())
+				.then().log()
+				.body().statusCode(200);
+
+		// check that the pays has been saved
+		Pays actual = repository.findOne(pays2.getId());
+		assertThat(actual.getId(), is(equalTo(pays2.getId())));
+		assertThat(actual.getNom(), is((equalTo(update.getNom()))));
+
+	}
```
Comment (Chombon) [inline backend/src/test/java/com/ambacam/rest/pays/PayssResourceIT.java]: Please check all properties

File: backend/src/main/java/com/ambacam/model/Pays.java [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+import org.hibernate.validator.constraints.NotEmpty;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+import java.io.Serializable;
+
+@Entity
+public class Pays implements Serializable {
+
+	private static final long serialVersionUID = -8019398584857375323L;
```
Comment (Chombon) [inline backend/src/main/java/com/ambacam/model/Pays.java]: Please remove this serialVersionUID (it is the same as Role) and generate its own serialVersionUID (put your mouse on the name of the class and then choose "add generate serial version ID")

**Comment by Chombon** [comment]
@MistraForest I am done with the review. You did two tickets in this branch(#34 and #21 ) so please remove all implementation related to the ticket #34 here and create his own branche. 
Please makes sure that the message of your commit meets the norm ( #ticketnummer ticketName) that is fixed next time.

File: backend/src/test/java/com/ambacam/rest/pays/PayssResourceIT.java (hunk 10 above)
Comment (MistraForest) [reply backend/src/test/java/com/ambacam/rest/pays/PayssResourceIT.java]: done

File: backend/src/test/java/com/ambacam/rest/pays/PayssResourceIT.java (hunk 9 above)
Comment (MistraForest) [reply backend/src/test/java/com/ambacam/rest/pays/PayssResourceIT.java]: done

File: backend/src/test/java/com/ambacam/rest/pays/PayssResourceIT.java (hunk 7 above)
Comment (MistraForest) [reply backend/src/test/java/com/ambacam/rest/pays/PayssResourceIT.java]: done

File: backend/src/main/java/com/ambacam/model/Pays.java [hunk 12]
```diff
@@ -0,0 +1,69 @@
+package com.ambacam.model;
+
+import org.hibernate.validator.constraints.NotEmpty;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+import java.io.Serializable;
+
+@Entity
+public class Pays implements Serializable {
```
Comment (Chombon) [inline backend/src/main/java/com/ambacam/model/Pays.java]: please generate the serialVersionUID (put your mouse on the name of the class and then choose "add generate serial version ID")

**Comment by Chombon** [comment]
@MistraForest  Please remove all implementation related to another ticket (#34 )

File: backend/src/main/java/com/ambacam/model/Pays.java (hunk 12 above)
Comment (MistraForest) [reply backend/src/main/java/com/ambacam/model/Pays.java]: This is not possible on intellij

File: backend/src/main/java/com/ambacam/service/ActionService.java [hunk 13]
```diff
@@ -61,24 +61,24 @@ public Action get(Long id) {
 	 * 
 	 * @param id The id of the action to update
 	 * 
-	 * @param action The new action modifications
+	 * @param update The new action modifications
```
Comment (Chombon) [inline backend/src/main/java/com/ambacam/service/ActionService.java]: Please remove all implementation related to another ticket (#34 )

File: backend/src/main/java/com/ambacam/service/ActionService.java [hunk 14]
```diff
@@ -61,24 +61,24 @@ public Action get(Long id) {
...
 	 * 
 	 * @return The action updated
 	 * 
 	 * @throws ResourceNotFoundException if the action is not found
 	 * @throws ResourceBadRequestException if an action with the name already exist
 	 */
-	public Action update(Long id, Action action) {
+	public Action update(Long id, Action update) {
```
Comment (Chombon) [inline backend/src/main/java/com/ambacam/service/ActionService.java]: Please remove all implementation related to another ticket (#34 )

File: backend/src/main/java/com/ambacam/service/ActionService.java [hunk 15]
```diff
@@ -61,24 +61,24 @@ public Action get(Long id) {
...
 
 		// find action
 		Action found = findAction(id);
 
-		if (!found.getNom().equals(action.getNom())) {
-			checkConsistency(action);
+		if (!found.getNom().equals(update.getNom())) {
```
Comment (Chombon) [inline backend/src/main/java/com/ambacam/service/ActionService.java]: Please remove all implementation related to another ticket (#34 )

File: backend/src/main/java/com/ambacam/service/ActionService.java [hunk 16]
```diff
@@ -61,24 +61,24 @@ public Action get(Long id) {
...
+			checkConsistency(update);
```
Comment (Chombon) [inline backend/src/main/java/com/ambacam/service/ActionService.java]: Please remove all implementation related to another ticket (#34 )

File: backend/src/main/java/com/ambacam/service/ActionService.java [hunk 17]
```diff
@@ -61,24 +61,24 @@ public Action get(Long id) {
...
 		}
 
-		action.setId(id);
-		return actionRepository.save(action);
+		update.setId(id);
```
Comment (Chombon) [inline backend/src/main/java/com/ambacam/service/ActionService.java]: Please remove all implementation related to another ticket (#34 )

File: backend/src/main/java/com/ambacam/service/ActionService.java [hunk 18]
```diff
@@ -61,24 +61,24 @@ public Action get(Long id) {
...
+		return actionRepository.save(update);
```
Comment (Chombon) [inline backend/src/main/java/com/ambacam/service/ActionService.java]: Please remove all implementation related to another ticket (#34 )

File: backend/src/main/java/com/ambacam/service/ActionService.java [hunk 19]
```diff
@@ -96,12 +96,12 @@ public void delete(Long id) {
 
 	private Action findAction(Long id) {
 		// find action
-		Action action = actionRepository.findOne(id);
```
Comment (Chombon) [inline backend/src/main/java/com/ambacam/service/ActionService.java]: Please remove all implementation related to another ticket (#34 )

File: backend/src/main/java/com/ambacam/service/ActionService.java [hunk 20]
```diff
@@ -96,12 +96,12 @@ public void delete(Long id) {
...
-		if (action == null) {
+		Action found = actionRepository.findOne(id);
+		if (found == null) {
 			throw new ResourceNotFoundException("The action " + id.toString() + " does not exist");
 		}
 
-		return action;
```
Comment (Chombon) [inline backend/src/main/java/com/ambacam/service/ActionService.java]: Please remove all implementation related to another ticket (#34 )

File: backend/src/main/java/com/ambacam/service/MotifSuppressionService.java [hunk 21]
```diff
@@ -74,9 +74,9 @@ public MotifSuppression get(Long id) {
 	public MotifSuppression update(Long id, MotifSuppression update) {
 
 		// find motifSuppression
-		MotifSuppression motifSuppression = findMotifSuppression(id);
```
Comment (Chombon) [inline backend/src/main/java/com/ambacam/service/MotifSuppressionService.java]: Please remove all implementation related to another ticket (#34 )

File: backend/src/main/java/com/ambacam/service/MotifSuppressionService.java [hunk 22]
```diff
@@ -74,9 +74,9 @@ public MotifSuppression get(Long id) {
...
+		MotifSuppression found = findMotifSuppression(id);
 
-		if (!motifSuppression.getNom().equals(update.getNom())) {
```
Comment (Chombon) [inline backend/src/main/java/com/ambacam/service/MotifSuppressionService.java]: Please remove all implementation related to another ticket (#34 )

File: backend/src/main/java/com/ambacam/service/MotifSuppressionService.java [hunk 23]
```diff
@@ -74,9 +74,9 @@ public MotifSuppression get(Long id) {
...
+		if (!found.getNom().equals(update.getNom())) {
```
Comment (Chombon) [inline backend/src/main/java/com/ambacam/service/MotifSuppressionService.java]: Please remove all implementation related to another ticket (#34 )

File: backend/src/main/java/com/ambacam/service/MotifSuppressionService.java [hunk 24]
```diff
@@ -101,12 +101,12 @@ public void delete(Long id) {
 
 	private MotifSuppression findMotifSuppression(Long id) {
 		// find motifSuppression
-		MotifSuppression motifSuppression = motifSuppressionRepository.findOne(id);
-		if (motifSuppression == null) {
+		MotifSuppression found = motifSuppressionRepository.findOne(id);
```
Comment (Chombon) [inline backend/src/main/java/com/ambacam/service/MotifSuppressionService.java]: Please remove all implementation related to another ticket (#34 )

File: backend/src/main/java/com/ambacam/service/MotifSuppressionService.java [hunk 25]
```diff
@@ -101,12 +101,12 @@ public void delete(Long id) {
...
+		if (found == null) {
 			throw new ResourceNotFoundException("The motifSuppression " + id.toString() + " does not exist");
 		}
 
-		return motifSuppression;
```
Comment (Chombon) [inline backend/src/main/java/com/ambacam/service/MotifSuppressionService.java]: Please remove all implementation related to another ticket (#34 )

File: backend/src/main/java/com/ambacam/service/RoleService.java [hunk 26]
```diff
@@ -72,9 +72,9 @@ public Role get(Long id) {
 	public Role update(Long id, Role update) {
 
 		// find role
-		Role role = findRole(id);
```
Comment (Chombon) [inline backend/src/main/java/com/ambacam/service/RoleService.java]: Please remove all implementation related to another ticket (#34 )

File: backend/src/main/java/com/ambacam/service/RoleService.java [hunk 27]
```diff
@@ -72,9 +72,9 @@ public Role get(Long id) {
 	public Role update(Long id, Role update) {
 
 		// find role
-		Role role = findRole(id);
+		Role found = findRole(id);
 
-		if (!role.getNom().equals(update.getNom())) {
```
Comment (Chombon) [inline backend/src/main/java/com/ambacam/service/RoleService.java]: Please remove all implementation related to another ticket (#34 )

File: backend/src/main/java/com/ambacam/service/RoleService.java [hunk 28]
```diff
@@ -99,12 +99,12 @@ public void delete(Long id) {
 
 	private Role findRole(Long id) {
 		// find role
-		Role role = roleRepository.findOne(id);
```
Comment (Chombon) [inline backend/src/main/java/com/ambacam/service/RoleService.java]: Please remove all implementation related to another ticket (#34 )

File: backend/src/main/java/com/ambacam/service/RoleService.java [hunk 29]
```diff
@@ -99,12 +99,12 @@ public void delete(Long id) {
...
 	private Role findRole(Long id) {
 		// find role
-		Role role = roleRepository.findOne(id);
-		if (role == null) {
+		Role found = roleRepository.findOne(id);
+		if (found == null) {
 			throw new ResourceNotFoundException("The role " + id.toString() + " does not exist");
 		}
 
-		return role;
```
Comment (Chombon) [inline backend/src/main/java/com/ambacam/service/RoleService.java]: Please remove all implementation related to another ticket (#34 )

File: backend/src/main/java/com/ambacam/service/StatusRequeteService.java [hunk 30]
```diff
@@ -72,9 +72,9 @@ public StatusRequete get(Long id) {
 	public StatusRequete update(Long id, StatusRequete update) {
 
 		// find statusRequete
-		StatusRequete statusRequete = findStatusRequete(id);
```
Comment (Chombon) [inline backend/src/main/java/com/ambacam/service/StatusRequeteService.java]: Please remove all implementation related to another ticket (#34 )

File: backend/src/main/java/com/ambacam/service/StatusRequeteService.java [hunk 31]
```diff
@@ -72,9 +72,9 @@ public StatusRequete get(Long id) {
 	public StatusRequete update(Long id, StatusRequete update) {
 
 		// find statusRequete
-		StatusRequete statusRequete = findStatusRequete(id);
+		StatusRequete found = findStatusRequete(id);
 
-		if (!statusRequete.getNom().equals(update.getNom())) {
```
Comment (Chombon) [inline backend/src/main/java/com/ambacam/service/StatusRequeteService.java]: Please remove all implementation related to another ticket (#34 )

File: backend/src/main/java/com/ambacam/service/StatusRequeteService.java [hunk 32]
```diff
@@ -99,12 +99,12 @@ public void delete(Long id) {
 
 	private StatusRequete findStatusRequete(Long id) {
 		// find statusRequete
-		StatusRequete statusRequete = statusRequeteRepository.findOne(id);
```
Comment (Chombon) [inline backend/src/main/java/com/ambacam/service/StatusRequeteService.java]: Please remove all implementation related to another ticket (#34 )

File: backend/src/main/java/com/ambacam/service/StatusRequeteService.java [hunk 33]
```diff
@@ -99,12 +99,12 @@ public void delete(Long id) {
...
 	private StatusRequete findStatusRequete(Long id) {
 		// find statusRequete
-		StatusRequete statusRequete = statusRequeteRepository.findOne(id);
-		if (statusRequete == null) {
+		StatusRequete found = statusRequeteRepository.findOne(id);
+		if (found == null) {
 			throw new ResourceNotFoundException("The statusRequete " + id.toString() + " does not exist");
 		}
 
-		return statusRequete;
```
Comment (Chombon) [inline backend/src/main/java/com/ambacam/service/StatusRequeteService.java]: Please remove all implementation related to another ticket (#34 )

**Comment by Chombon** [comment]
@MistraForest  You have to rename these parameters as they were at the base. Because the renaming of those is done in another ticket.

File: backend/src/main/java/com/ambacam/model/Pays.java (hunk 12 above)
Comment (Chombon) [reply backend/src/main/java/com/ambacam/model/Pays.java]: Have you tried this?
put your mouse on the name of the class and then choose "add generate serial version ID"

Comment (Chombon) [reply backend/src/main/java/com/ambacam/model/Pays.java]: If it does not work. Here is a link that explains you how to activate and generate it:
https://netlicensing.io/blog/2012/12/25/intellij-idea-how-to-generate-serialversionuid/

**Pull request merged**